tests/kernel/run_ltp.pm

```diff
@@ -332,6 +332,10 @@ sub pre_run_hook {
     # But change them to hard fail in this test module.
     for my $pattern (@{$self->{serial_failures}}) {
         my %tmp = %$pattern;
+
+        # don't switch to hard fail when test is expected to produce kernel warning
+        next if ($tmp{soft_on_expect_warn} =~ m/true/i && get_var('LTP_WARN_EXPECTED'));
```

Comment:
You could set `soft_on_expect_warn => 1` and get rid of the regex here. Also, matching `undef =~ m/.../` triggers perl warning.